Further button functionality to Drive_2_0_fieldCentric.java

gwarby opened the pull request:
Added code to use buttons to grab different levels of pixels/return to holding. Used positions from BlueBackPixel code. Major edits to lines 245-257

gwarby committed:
Further button functionality to Drive_2_0_fieldCentric.java

Added code to use buttons to grab different levels of pixels/return to holding

lcdnbl commented on Drive_2_0_fieldCentric.java:
```diff
@@ -237,6 +241,20 @@ public void runOpMode() {
           armraise.setPower(driverCmd_ArmRaise * MAX_ARM_LOWER_POWER);
         }
       }
+      
+      // run arm to set positions using buttons
+      // a -> ground, b -> second stacked pixel, y -> holding
+      // x grabs/ lets go of pixel (still)
+      if (DriverCmd_ArmToGround) {
+        isArmHolding = true;
+        armraise.setTargetPosition(0);
+      } else if (DriverCmd_GrabTopPixel) {
+        isArmHolding = true;
+        armraise.setTargetPosition(4.557);
+      } else if (DriverCmd_ArmToHolding) {
+        isArmHolding = false;
+        armraise.setTargetPosition(MIN_ARM_RAISE_TICKS_WHEN_NOT_PICKING_UP);
+      }
```
I think instead of `armraise.setTargetPosition(<something>);` 
you want `armRaiseTargetPosition = <something>;`
This will then fall into / use the code from line 203-216 (on the next loop) to drive to this variable.
I'm concerned these code targets will "fight" with the code on those lines otherwise.

lcdnbl commented on Drive_2_0_fieldCentric.java:
```diff
@@ -237,6 +241,20 @@ public void runOpMode() {
+        isArmHolding = true;
+        armraise.setTargetPosition(4.557);
+      } else if (DriverCmd_ArmToHolding) {
+        isArmHolding = false;
```
I think this should be `isArmHolding = true` ?

gwarby committed:
Update Drive_2_0_fieldCentric.java

Changed to use function instead of giving direct command in ticks

gwarby committed:
Drone controls added to Drive_2_0_fieldCentric.java

Description on line 375. Possibly add automatic line up/shot soon?

gwarby committed:
Change button functions for Drive_2_0_fieldCentric.java

Lowered 'grab top pixel' height, per testing with autonomous

gwarby committed:
Drone launcher bug fix Drive_2_0_fieldCentric.java

Corrected name

gwarby committed:
Feat: only buttons go to ground. fix: buttons put arm in holding state Drive_2_0_fieldCentric.java

The stick no longer will get the arm past the limit, only buttons will. Buttons also now put arm into holding state
